src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.NotebookTests/Utils.cs

```diff
@@ -0,0 +1,134 @@
+                process.Start();
+                process.WaitForExit();
+            }
+        }
+
+        public static Process SetupNotebookProcessRun(string notebookPath, string? outputNotebookPath = null)
+        {
+            Process process = new Process();
+            process.StartInfo.FileName = "dotnet-repl";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.Arguments = $"--run {notebookPath} --exit-after-run";
+
+            // Optionally include the output notebook path in case we care about the output.
+            if (!string.IsNullOrEmpty(outputNotebookPath))
+            {
+                process.StartInfo.Arguments += $" --output-path {outputNotebookPath}";
+            }
+
+            // The working directory should be the directory that the notebook is in.
+            process.StartInfo.WorkingDirectory = Path.GetDirectoryName(notebookPath);
+            return process;
+        }
+
+        public static void RunNotebookThatsExpectedToPass(string notebookPath) 
+        {
+            string tempPathForOutputNotebook = Path.GetTempFileName();
+            using (Process dotnetReplProcess = SetupNotebookProcessRun(notebookPath, tempPathForOutputNotebook))
+            {
+                dotnetReplProcess.Start();
+                dotnetReplProcess.WaitForExit(TIMEOUT);
+
+                // If notebook execution fails, parse the notebook and find the error.
+                if (dotnetReplProcess.ExitCode != 0)
+                {
+                    ParseNotebookAndFindErrors(tempPathForOutputNotebook, notebookPath);
+                }
+
+                else
+                {
+                    dotnetReplProcess.HasExited.Should().BeTrue();
+                }
+            }
+
+            // Cleanup any temp notebooks.
+            File.Delete(tempPathForOutputNotebook);
+        }
+
+        public static void RunNotebookThatsExpectedToFail(string notebookPath)
+        {
+            string tempPathForOutputNotebook = Path.GetTempFileName();
+            using (Process dotnetReplProcess = SetupNotebookProcessRun(notebookPath, tempPathForOutputNotebook))
+            {
+                dotnetReplProcess.Start();
+                dotnetReplProcess.WaitForExit(TIMEOUT);
+                dotnetReplProcess.ExitCode.Should().NotBe(0);
+            }
+        }
+
+        public static NotebookRoot? ParseNotebook(string notebookPath)
+        {
+            string failedNotebookText = File.ReadAllText(notebookPath);
+            NotebookRoot? deserializedNotebook = JsonConvert.DeserializeObject<NotebookRoot>(failedNotebookText);
+            deserializedNotebook.Should().NotBeNull();
+            return deserializedNotebook;
+        }
+
+        public static void ParseNotebookAndFindErrors(string outputNotebookPath, string baseNotebookPath)
+        {
+            NotebookRoot? deserializedNotebook = ParseNotebook(outputNotebookPath);
+            File.Delete(outputNotebookPath);
```

Comment:
`Delete` doesn't belong in a `Parse` method.  The caller is also already deleting the file.